src/components/ha-sidebar.ts

```diff
@@ -198,934 +41,88 @@ class HaSidebar extends SubscribeMixin(LitElement) {
-    await this.updateComplete;
-  }
-
-  private async _createSortable() {
-    const Sortable = await loadSortable();
-    this._sortable = new Sortable(
-      this.shadowRoot!.getElementById("sortable")!,
-      {
-        animation: 150,
-        fallbackClass: "sortable-fallback",
-        dataIdAttr: "data-panel",
-        handle: "paper-icon-item",
-        onSort: async () => {
-          this._panelOrder = this._sortable!.toArray();
-        },
-      }
-    );
-  }
-
-  private _deactivateEditMode() {
-    this._sortable?.destroy();
-    this._sortable = undefined;
-  }
-
-  private _closeEditMode() {
-    fireEvent(this, "hass-edit-sidebar", { editMode: false });
-  }
-
-  private async _hidePanel(ev: Event) {
-    ev.preventDefault();
-    const panel = (ev.currentTarget as any).panel;
-    if (this._hiddenPanels.includes(panel)) {
-      return;
-    }
-    // Make a copy for Memoize
-    this._hiddenPanels = [...this._hiddenPanels, panel];
-    this._renderEmptySortable = true;
-    await this.updateComplete;
-    const container = this.shadowRoot!.getElementById("sortable")!;
-    while (container.lastElementChild) {
-      container.removeChild(container.lastElementChild);
-    }
-    this._renderEmptySortable = false;
-  }
-
-  private async _unhidePanel(ev: Event) {
-    ev.preventDefault();
-    const panel = (ev.currentTarget as any).panel;
-    this._hiddenPanels = this._hiddenPanels.filter(
-      (hidden) => hidden !== panel
-    );
-    this._renderEmptySortable = true;
-    await this.updateComplete;
-    const container = this.shadowRoot!.getElementById("sortable")!;
-    while (container.lastElementChild) {
-      container.removeChild(container.lastElementChild);
-    }
-    this._renderEmptySortable = false;
-  }
-
-  private _itemMouseEnter(ev: MouseEvent) {
-    // On keypresses on the listbox, we're going to ignore mouse enter events
-    // for 100ms so that we ignore it when pressing down arrow scrolls the
-    // sidebar causing the mouse to hover a new icon
-    if (
-      this.alwaysExpand ||
-      new Date().getTime() < this._recentKeydownActiveUntil
-    ) {
-      return;
-    }
-    if (this._mouseLeaveTimeout) {
-      clearTimeout(this._mouseLeaveTimeout);
-      this._mouseLeaveTimeout = undefined;
-    }
-    this._showTooltip(ev.currentTarget as PaperIconItemElement);
   }
 
-  private _itemMouseLeave() {
-    if (this._mouseLeaveTimeout) {
-      clearTimeout(this._mouseLeaveTimeout);
-    }
-    this._mouseLeaveTimeout = window.setTimeout(() => {
-      this._hideTooltip();
-    }, 500);
+  private _panelHover(ev: CustomEvent) {
+    this._mouseOverItem({ target: ev.detail });
   }
 
-  private _listboxFocusIn(ev) {
-    if (this.alwaysExpand || ev.target.nodeName !== "A") {
+  private _mouseOverItem(ev: { target: HTMLElement }) {
+    const tooltipRoot =
+      this.shadowRoot?.querySelector("ha-sidebar-tooltip")?.shadowRoot;
+    const tooltip = tooltipRoot?.firstElementChild as HTMLElement;
```

Comment:
Why not just use a `slot`?

Reply:
i don't think a slot would make things easier from what i understand, the tooltip would still be in a shadow root, and you would require even more logic to traverse the slot